### PR TITLE
morebits: api improvements

### PR DESCRIPTION
**Pull request by siddharthvp**
#1029 seems rather far-fetched for now, so it might still be worth making some improvements to Morebits.wiki.api.

### CODE REVIEW

File: morebits.js [hunk 1]
```diff
@@ -1774,7 +1777,7 @@ Morebits.wiki.api.prototype = {
 
 		var ajaxparams = $.extend({}, {
 			context: this,
-			type: 'POST',
+			type: this.query.action === 'query' ? 'GET' : 'POST',
```
Comment (Amorymeltzer) [inline morebits.js:1780]: Nice

File: morebits.js [hunk 2]
```diff
@@ -1721,6 +1721,8 @@ Morebits.wiki.api = function(currentAction, query, onSuccess, statusElement, onE
 	}
 	if (!query.format) {
 		this.query.format = 'xml';
+	} else if (query.format === 'json' && !query.formatversion) {
+		this.query.formatversion = '2';
```
Comment (Amorymeltzer) [inline morebits.js:1725]: Good point about this being a good time.  I'm already using it locally, so 👍

File: morebits.js [hunk 3]
```diff
@@ -1738,6 +1740,7 @@ Morebits.wiki.api.prototype = {
 	statusText: null, // result received from the API, normally "success" or "error"
 	errorCode: null, // short text error code, if any, as documented in the MediaWiki API
 	errorText: null, // full error description, if any
+	tokenRetry: false,
```
Comment (Amorymeltzer) [inline morebits.js:1743]: Looks like you renamed this to `badtokenRetry` below (or vice-versa)

Comment (siddharthvp) [reply morebits.js]: Indeed. So what messed me up here was the later realisation that all actions don't use a CSRF token. If we want to be fussy about it, we could do something like [what I do in my bot framework](https://github.com/siddharthvp/mwn/blob/master/src/bot.js#L525) (querying the type of token needed as well) but IMO not really necessary. 

Easier to pretend that everything is CSRF-based (badtoken error on anything which isn't will be failing as before).

Comment (Amorymeltzer) [reply morebits.js]: Indeed, that's roughly where I ended up thinking.  It's few and far between atm, so nbd.

**Pull request merged**